testsuite/features/support/file_management.rb

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+# Attempts to retrieve the SHA256 checksum for a file inside a given directory or download it
+# from the same domain the file has been downloaded from.
+# Returns the path to the checksum or nil if none has been found/downloaded
+def get_checksum_path(dir, original_file_name, file_url)
+  checksum_file_names = %W[CHECKSUM SHA256SUMS sha256sum.txt #{original_file_name}.CHECKSUM #{original_file_name}.sha256]
+
+  server = get_target('server')
+  # When using a mirror, the checksum file should be present and in the same directory of the file
+  if $mirror
+    checksum_file_names.each do |name|
+      checksum_path = "#{dir}/#{name}"
+      _output, code = server.run("test -e #{checksum_path}", check_errors: false)
+      return checksum_path if code.zero?
+    end
+  # Attempt to download the checksum file
+  else
+    base_url = file_url.delete_suffix(original_file_name)
+
+    checksum_file_names.each do |name|
+      checksum_url = base_url + name
+      _output, code = server.run("cd #{dir} && wget --no-check-certificate #{checksum_url}", check_errors: false, timeout: 1500)
```

Comment:
Instead of trying every item in this list, as a call to the server, we will then try to download it, and finally, we process if it's ok or not.
What about, from the controller, we perform HTTP request to the different possible URLs, identifying which of them is the valid URL for this original_file_name.
And after that, we only need to make a unique call to the server to download from the valid URL, and if it fails... that we raise an exception.

Reply:
Btw, be cautious with the value on the timeout parameter, if for some reason the server call gets stuck, we will be waiting idly for 1500 seconds, and the total time of the test suite will increase a lot.